### ISSUE TITLE
Dummy provisioner should support Pre-provisioned hosts

### ISSUE THREAD

**Issue by jaypoulz** [CentOS-PaaS-SIG/linchpin#738]
# Overview
I have a use case where I own a machine that lives somewhere. Say `my-test-host.redhat.com`.
I want to use linchpin to create an inventory file for this host using my custom layout, without actually provisioning it (as it is already provisioned).

Right now (by leaving uhash enabled), I can get pretty close using the dummy provisioner.
## Sample Output
```
[all]
my-test-host.redhat.com-c4e8ea-0.example.net
```
I accomplished this by overriding the `name` key in the dummy resource definition dynamically via template-data.

Ideally, I want to be able to force the dummy provisioner to use my passed hostname *as is*, without appending the uhash or the fake domain.

## Desired Output
```
[all]
my-test-host.redhat.com
```

**Comment by herlo** [comment]
@jaypoulz this feels very much like what is discussed [here](https://github.com/CentOS-PaaS-SIG/linchpin/issues/486#issuecomment-371584451). Specifically in the 'vars' (now called cfgs) section, under the 'evars' subsection. We've done some initial work for this with the cfgs in the [PinFile](https://github.com/CentOS-PaaS-SIG/linchpin/pull/703). In this case, however, the code might look something like this:

```
cfgs:
  evars:
    dummy_tld: 'redhat.com'
```

In this case, the evar would actually be named `{{ dummy_tld }}` in the ansible playbook where this would need to be changed. I wonder if this would provide the functionality you are looking for?

Mostly, this idea came from the original work you and I discussed very long ago to accommodate #448 (which brought #486 into existence).

**Comment by samvarankashyap** [comment]
@jaypoulz @herlo : How about adding a domain option to dummy provider to override existing domain.? Anythoughts ?

**Comment by jaypoulz** [comment]
@herlo I don't think overriding the domain at a config level is sufficient, since I could have several resources in the resource_defs with different domains. It would be fine to have that as a default to be overridden, but I think SK's proof of concept is more powerful because domains are specified at the schema level.

I made an additional comment about per topology uhash. See #742 for details.

**Comment by samvarankashyap** [comment]
fixed by #742
